EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInput.java

```diff
@@ -0,0 +1,282 @@
+		return selectedMethod;
+	}
+
+	/**
+	 * Setter for the currently selected method.
+	 *
+	 * @param selectedMethod that should be set
+	 */
+	public void setSelectedMethod(ProbabilityDistributionEnum selectedMethod) {
+		this.selectedMethod = selectedMethod;
+	}
+
+	/**
+	 * Getter for the current input.
+	 *
+	 * @return the current input
+	 */
+	public String getInput() {
+		return input;
+	}
+
+	/**
+	 * Returns the current input, in case this is empty, a default value defined by the class is returned.
+	 *
+	 * @return a string containing the current input or the default input
+	 */
+	String getInputOrDefault() {
+		if(input != null && !input.isEmpty()) {
+			return input;
+		}
+		return this.getDefaultInput();
+	}
+
+	/**
+	 * Setter for the input.
+	 *
+	 * @param input to be set
+	 */
+	public void setInput(String input) {
+		this.input = input;
+	}
+
+	/**
+	 * Returns the chosen value from user input. If no value was chosen up to now, this will be done first.
+	 *
+	 * @return a serializable object containing a single value chosen from user input
+	 */
+	public Serializable getCalculatedValue() {
+		if (this.getValue() == null) {
+			this.calculateRandomValue();
+		}
+		return this.getValue();
+	}
+
+	/**
+	 * Chooses a value from the user input and saves it in the value-instance variable.
+	 *
+	 */
+	public abstract void calculateRandomValue();
+
+	/**
+	 * Abstract getter for the value instance variable defined in subclasses.
+	 * Will be overridden in subclasses.
+	 *
+	 * @return a serializable object
+	 */
+	abstract Serializable getValue();
+
+	/**
+	 * Returns the chosen value as string instead of serializable object.
+	 * It will execute the calculation of a value in case this wasn't done until now.
+	 * Might be overriden in case the parsing to string isn't this easy (e.g. in DateAttributeInput).
+	 *
+	 * @return a string
+	 */
+	public String getValueAsString() {
+		return String.valueOf(this.getCalculatedValue());
+	}
+
+	/**
+	 * Returns the default value, that should be used if the user input is empty.
+	 * Will be overridden in subclasses.
+	 *
+	 * @return a default value as string
+	 */
+	String getDefaultInput() { return "UNDEFINED"; }
+
+	/**
+	 * Checks whether the user input is contained in a given range.
+	 * Might get overridden in subclass.
+	 *
+	 * @param range to be searched in
+	 * @return bool if string is in range
+	 */
+	public boolean isInRange(String range) {
+		String[] possibleValues = range.split(";");
+		for (String possibleValue : possibleValues) {
+			if (possibleValue.equals(this.getValueAsString())) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Returns the object as a string containing the most important information concerning the instance.
+	 *
+	 * @return a string describing the AttributeInput-Object
+	 */
+	@Override
+	public String toString() {
+		StringBuilder stringify = new StringBuilder(System.getProperty("line.separator"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "line.separator" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)